Custom scalar is mocked as a type

My schema specifies two scalars:
```
scalar Date
scalar AnyObject
input ServiceParametersInput {
  date: Date!
  serviceName: String!
  data: AnyObject
}
```

However, the  `AnyObject` field appears to be mocked as an actual type, without any mock function being generated for it, whereas the `Date` field is mocked "as expected" - although it's also presented as a typical custom scalar in the [graphql doc](https://graphql.org/learn/schema/#scalar-types):
```
export const aServiceParametersInput = (overrides?: Partial<ServiceParametersInput>): ServiceParametersInput => {
    return {
        date: overrides && overrides.hasOwnProperty('date') ? overrides.date! : '1970-01-13T19:39:08.227Z',
        serviceName: overrides && overrides.hasOwnProperty('serviceName') ? overrides.serviceName! : 'praesentium',
        data: overrides && overrides.hasOwnProperty('data') ? overrides.data! : anAnyObject(),
    };
};
```

This results into a compile-time error when using the generated mocks file:
```
    schema/lib/mocks.tsx:221:81 - error TS2304: Cannot find name 'anAnyObject'.

    221         data: overrides && overrides.hasOwnProperty('data') ? overrides.data! : anAnyObject(),
                                                                                        ~~~~~~~~~~~
```

For now I'm working around this by adding a dummy definition:
```
  ./schema/lib/mocks.tsx:
    plugins:
      - "graphql-codegen-typescript-mock-data":
          typesFile: "./clientTypes.tsx"
          addTypename: true
      # work around issue with mocking of custom scalars
      - add: const anAnyObject = ():any => null;
```

Thanks.

--- a/src/index.ts
+++ b/src/index.ts
@@ -100,6 +100,10 @@ const getNamedType = (
                             enumValuesConvention,
                             foundType.types && foundType.types[0],
                         );
+                    case 'scalar':
+                        // it's a scalar, let's use a string as a value.
+                        // This could be improved with a custom scalar definition in the config
+                        return `'${casual.word}'`;
                     default:
                         throw `foundType is unknown: ${foundType.name}: ${foundType.type}`;
                 }
@@ -175,7 +179,7 @@ export interface TypescriptMocksPluginConfig {
 
 interface TypeItem {
     name: string;
-    type: string;
+    type: 'enum' | 'scalar' | 'union';
     values?: string[];
     types?: readonly NamedTypeNode[];
 }
@@ -278,6 +282,15 @@ export const plugin: PluginFunction<TypescriptMocksPluginConfig> = (schema, docu
                 },
             };
         },
+        ScalarTypeDefinition: (node) => {
+            const name = node.name.value;
+            if (!types.find((enumType) => enumType.name === name)) {
+                types.push({
+                    name,
+                    type: 'scalar',
+                });
+            }
+        },
     };
 
     const result: any = visit(astNode, { leave: visitor });
@@ -286,7 +299,7 @@ export const plugin: PluginFunction<TypescriptMocksPluginConfig> = (schema, docu
     const typeImports = definitions
         .map(({ typeName }: { typeName: string }) => typeName)
         .filter((typeName: string) => !!typeName);
-    typeImports.push(...types.map(({ name }) => name));
+    typeImports.push(...types.filter(({ type }) => type !== 'scalar').map(({ name }) => name));
     // List of function that will generate the mock.
     // We generate it after having visited because we need to distinct types from enums
     const mockFns = definitions.map(({ mockFn }: any) => mockFn).filter((mockFn: Function) => !!mockFn);

--- a/tests/typescript-mock-data.spec.ts
+++ b/tests/typescript-mock-data.spec.ts
@@ -4,6 +4,9 @@ import { buildSchema } from 'graphql';
 import { plugin } from '../src';
 
 const testSchema = buildSchema(/* GraphQL */ `
+    scalar Date
+    scalar AnyObject
+
     type Avatar {
         id: ID!
         url: String!
@@ -11,10 +14,12 @@ const testSchema = buildSchema(/* GraphQL */ `
 
     type User {
         id: ID!
+        creationDate: Date!
         login: String!
         avatar: Avatar
         status: Status!
         customStatus: ABCStatus
+        scalarValue: AnyObject!
     }
 
     type Query {
@@ -53,6 +58,16 @@ it('should generate mock data functions', async () => {
     const result = await plugin(testSchema, [], {});
 
     expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
+
+it('should generate mock data functions with scalars', async () => {
+    const result = await plugin(testSchema, [], {});
+
+    expect(result).toBeDefined();
+    expect(result).toContain(
+        "scalarValue: overrides && overrides.hasOwnProperty('scalarValue') ? overrides.scalarValue! : 'neque',",
+    );
     expect(result).toMatchSnapshot();
 });
 

